### PR TITLE
Build structure, first step

### PR DESCRIPTION
**Pull request by RayRoestenburg**

### CODE REVIEW

**Comment by RayRoestenburg** [comment]
There are some deps from java-sdk to core, Effect and other types, so first just moving everything to sdkJava, and get the new build structure in place (more sub projects), copy pasted quite a bit, which can be fixed over time.

**Comment by RayRoestenburg** [comment]
Requesting review a little early, would like to get this in quickly, since the moves are quite large..

File: build.sbt [hunk 1]
```diff
@@ -2,10 +2,46 @@ import Dependencies._
...
+  .aggregate(sdkCore, sdkJava, sdkScala, testkitJava, tck, codegenCore, codegenJava, codegenJavaCompilationTest)
 
-lazy val sdk = project
-  .in(file("sdk"))
+//FIXME duplicating most settings in sdkCore, sdkJava, sdkScala for now, cleanup later.
+lazy val sdkCore = project
+  .in(file("sdk/core"))
+  .enablePlugins(AkkaGrpcPlugin, BuildInfoPlugin, PublishSonatype)
+  .settings(
+    name := "akkaserverless-java-sdk-core",
```
Comment (raboof) [inline build.sbt:12]: maybe `akkaserverless-jvm-sdk` or `akkaserverless-runtime`?

File: build.sbt [hunk 2]
```diff
@@ -2,10 +2,46 @@ import Dependencies._
...
+    crossPaths := false,
+    buildInfoKeys := Seq[BuildInfoKey](
```
Comment (raboof) [inline build.sbt:14]: do we need `BuildInfo` in all 3 artifacts?

File: build.sbt [hunk 3]
```diff
@@ -2,10 +2,46 @@ import Dependencies._
...
+      val javaSourceDir = (Compile / javaSource).value.getAbsolutePath
+      (Compile / doc / sources).value.filter(_.getAbsolutePath.startsWith(javaSourceDir))
+    },
+    // javadoc (I think java 9 onwards) refuses to compile javadocs if it can't compile the entire source path.
+    // but since we have java files depending on Scala files, we need to include ourselves on the classpath.
+    Compile / doc / dependencyClasspath := (Compile / fullClasspath).value,
+    Compile / compile / javacOptions ++= Seq("--release", "8"),
+    Compile / scalacOptions ++= Seq("-release", "8"),
+    Compile / akkaGrpcGeneratedSources := Seq(AkkaGrpc.Server),
+    Compile / akkaGrpcGeneratedLanguages := Seq(AkkaGrpc.Scala), // FIXME should be Java, but here be dragons
```
Comment (raboof) [inline build.sbt:33]: do we need Akka gRPC in all projects/configurations?

**Review by raboof — APPROVED** [review]
good to get the initial structure in 'early' to avoid conflicts, but let's not wait too long with cleaning up further ;)

File: build.sbt (hunk 1 above)
Comment (RayRoestenburg) [reply build.sbt:12]: I like the jvm-sdk suggestion

File: build.sbt (hunk 2 above)
Comment (RayRoestenburg) [reply build.sbt:14]: I'm not sure, first step was to copy paste, and remove later. I can remove them from core. I guess they should be there for scala and java sdk.

File: build.sbt (hunk 3 above)
Comment (RayRoestenburg) [reply build.sbt:33]: I'll remove this from core now, though I need to look deeper into where stuff is generated etc.

Comment (RayRoestenburg) [reply build.sbt:33]: (Since there is nothing in core yet, can always bring it back when needed)

Comment (raboof) [reply build.sbt]: it looks like this is the server that the sdk provides to the proxy, so perhaps this should later move along with the implementation to core

**Pull request merged**